controllers/periodic_syncer_test.go

```diff
@@ -182,4 +198,68 @@ func Test_Sync_Skip_Existing(t *testing.T) {
 		{Name: "bar"},
 	}, newMemb.Spec.UserRefs)
 
+	newTeam := controlv1.Team{}
+	require.NoError(t, c.Get(ctx, types.NamespacedName{Namespace: "foo", Name: "bar"}, &newTeam))
+	assert.ElementsMatch(t, []controlv1.UserRef{
+		{Name: "baz"},
+		{Name: "qux"},
+	}, newTeam.Spec.UserRefs)
+}
+
+func Test_Sync_Skip_ExistingUsers(t *testing.T) {
+	ctx := context.Background()
+	subject := controlv1.User{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "existing",
+		},
+		Spec: controlv1.UserSpec{
+			Preferences: controlv1.UserPreferences{
+				DefaultOrganizationRef: "current-organization",
+			},
+		},
+	}
+
+	c, keyMock, _ := prepareTest(t, fooOrg, fooMemb, &subject)
+
+	fooGroup := keycloak.NewGroup("foo")
+	fooGroup.Members = []keycloak.User{
+		{
+			Username:               subject.Name,
+			DefaultOrganizationRef: "updated-organization",
+		},
+	}
+
+	keyMock.EXPECT().
+		ListGroups(gomock.Any()).
+		Return([]keycloak.Group{fooGroup}, nil).
+		Times(1)
+
+	err := (&PeriodicSyncer{
+		Client:   c,
+		Keycloak: keyMock,
+	}).Sync(ctx)
+	require.NoError(t, err)
+
+	updatedSubject := controlv1.User{}
+	require.NoError(t, c.Get(ctx, types.NamespacedName{Name: subject.Name}, &updatedSubject))
+	assert.Equal(t, subject, subject)
```

Comment:
I assume this should be 

```suggestion
	assert.Equal(t, subject, updatedSubject)
```

otherwise the assertion will always be equal.

Reply:
Ouch 🙈